### ISSUE TITLE
SCM: editor selection not applied when jumping from diff back to editor

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#30674]
We now nicely restore the selection when jumping from editor to the diff editor but we do not seem to do the same thing when going from diff editor to normal editor (which is actually my main use case).

I think the missing piece is that `vscode.open` allows to provide the same editor options and git extension to adopt that. Created PR https://github.com/Microsoft/vscode/pull/30675 for the missing API in the command.

**Comment by joaomoreno** [comment]
I wonder how long it would take for you to notice this.

**Comment by bpasero** [comment]
@joaomoreno you do not really depend on #30829 imho, you can always do `openTextDocument` followed by `showTextDocument`. My PR is just syntactic sugar for that.

**Comment by joaomoreno** [comment]
Working.

**Comment by bpasero** [comment]
Now it is...

**Comment by bpasero** [comment]
Actually I found another case where it was not working: When having a file on the index.

**Verify:** you can jump between file and diff by maintaining the selection and it works if a change is staged or not.

**Comment by dbaeumer** [comment]
Needs to only work from diff -> editor. Verified. The other direction is not working. Hope that is expected.

**Comment by bpasero** [comment]
@dbaeumer it should work in both directions both if a file is in "Working Files" as well as "Index".

**Comment by dbaeumer** [comment]
@bpasero the other direction didn't work for me. It always selected the first diff regardless of what I had selected before. Leave it up to you and @joaomoreno to reopen or to create a new issue.

**Comment by joaomoreno** [comment]
It depends on where the command is executed from:

- Clicking a diff in the viewlet: reveals the first diff
- Choosing the `Open Change` context menu command in the viewlet: reveals the first diff
- Editor title: keeps selection
- Command Pallette: keeps selection
